### ISSUE TITLE
Feedback on ROM - Queries

### ISSUE THREAD

**Issue by janko-m** [rom-rb/rom-rb.org#227]
First of all, the documentation for ROM is excellent! 👏 (both RDocs and guides)

The "[Projecting function results]" section inside the "SQL - Queries" guide shows adding function calls to the SQL query using this syntax:

```rb
int::count(...)
string::concat(...)
```

However, I couldn't find any clarification on this neither here nor in the `rom-sql` RDocs. I know that that this syntax is used for the SQL builder to know when you're adding a function call and when a column identifier (as both are method calls). But it would be great to have an explanation somewhere what are the typed `int::` and `string::` "namespaces" for, as I'm wondering why isn't there just a single `function::` "namespace" 😃

**Comment by solnic** [comment]
Right, this should be documented. The reason why we require return type, is because we're inferring structs automatically, and we want to know both attribute names and their types.

**Comment by solnic** [comment]
One more thing, we have `Any` type, so it is possible to add a generic function builder that doesn't require return type, like the `function::` syntax you proposed. WDYT @flash-gordon?

**Comment by flash-gordon** [comment]
@solnic sounds like a nice idea. I think we could have `fun` for brevity.

**Comment by solnic** [comment]
👉🏻 https://github.com/rom-rb/rom-sql/issues/270 👈🏻
